spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/DefaultPollableMessageSource.java

```diff
@@ -277,7 +278,12 @@ protected boolean shouldRequeue(Exception e) {
 	 * Receives Message from the source and converts its payload to a provided type.
 	 * Can return null
 	 */
-	private Message<?> receive(ParameterizedTypeReference<?> type) {
+	@Override
+	public Message<?> receive(ParameterizedTypeReference<?> type) {
+		//Prevent a race condition if this is called before setSource().
+		if (this.source == null) {
+			return null;
+		}
```

Comment:
How can that be possible? The poller won't start until all beans have been populated.

Reply:
I thought that too, but there is definitely a race condition.  I was seeing a couple of NPEs at first when running the test.